### PR TITLE
Quadlet: add support for setting --ip and --ip6

### PR DESCRIPTION
**Pull request by restitux**
Add support for `IP=` and `IP6=`.

```release-note
Added support for specifying static IPv4 and IPv6 addresses in Quadlet .container files via the IP= and IP6= options.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
I don't believe any of the logs belong in here?
test/e2e/ginkgo-node-1.log

File: test/e2e/ginkgo-node-1.log
Comment (vrothberg) [inline test/e2e/ginkgo-node-1.log:1]: Please remove this log and the other .log files.

**Review by vrothberg — COMMENTED** [review]
Thanks for contributing!

File: docs/source/markdown/podman-systemd.unit.5.md
Comment (ygalblum) [reply docs/source/markdown/podman-systemd.unit.5.md:209]: Not sure what I think. On one hand I agree that using `IPv4` is more obvious. But on the other, it translate to `--ip`

**Comment by ygalblum** [comment]
I think you're missing an empty line between the commit's title and the signoff comment.

**Review by Luap99 — COMMENTED** [review]
I would not recommend using --ip or --ipv6, they do not work when using more than one network.

I haven't actually tested this but quadlet already supports `Network=` and the cli syntax to properly set static ip/ipv6/mac is this: `network:ip=192.168.1.10`. This allows you to set a static ip for each network that is specified. The options are described here: https://docs.podman.io/en/latest/markdown/podman-create.1.html#network-mode-net

This was added with podman 4.0, the --ip and --ip6 options mostly exists because of backwards/docker compat.
I will leave it to the quadlet maintainers to decide if they want these options but I say the new syntax is much better because it support all network options.

File: docs/source/markdown/podman-systemd.unit.5.md
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-systemd.unit.5.md:209]: We've already established `--ip` for IPv4 and `--ip6` for IPv6 in the run command [run Man page](https://docs.podman.io/en/latest/markdown/podman-run.1.html), so I think we should stick with those.

**Comment by TomSweeneyRedHat** [comment]
@ygalblum looks like you need to sign this.
Other than the logs that @vrothberg pointed out, LGTM

File: test/e2e/ginkgo-node-1.log
Comment (restitux) [reply test/e2e/ginkgo-node-1.log]: Just removed these. I also added a `.gitignore` rule for them

**Comment by restitux** [comment]
@Luap99 using the `ip` parameter on a network via `Network=` does in fact work with quadlet, that's actually what I'm using in lieu of this PR.

**Comment by restitux** [comment]
> I think you're missing an empty line between the commit's title and the signoff comment.

The signoff should be fixed now, I made a new commit when I deleted the logs. LMK if there are still issues

**Comment by ygalblum** [comment]
@TomSweeneyRedHat I was waiting for the decision regarding the name.
/lgtm
/hold

**Review by ygalblum — APPROVED** [review]

**Comment by rhatdan** [comment]
@ygalblum Could you respond to @Luap99 about this?  Should we have two ways to handle this?  Especially when we sort of deprecate the ip and ip6 way?

**Comment by ygalblum** [comment]
@rhatdan @Luap99 I don't see any harm in supporting in Quadlet whatever Podman already supports. In addition, I do see a benefit in supporting and using simple flags whenever possible. I understand the issue with multiple networks, but, I think that still most cases use a single one.

**Pull request merged**
